src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -4570,4 +4597,168 @@ namespace winrt::TerminalApp::implementation
+        if (tabImpl)
+        {
+            // First: stash the tab we started dragging.
+            // We're going to be asked for this.
+            _stashedDraggedTab = tabImpl;
+
+            // Into the DataPackage, let's stash our own window ID.
+            const winrt::hstring id{ fmt::format(L"{}", _WindowProperties.WindowId()) };
+
+            // Get our PID
+            const winrt::hstring pid{ fmt::format(L"{}", GetCurrentProcessId()) };
+
+            e.Data().Properties().Insert(L"windowId", winrt::box_value(id));
+            e.Data().Properties().Insert(L"pid", winrt::box_value(pid));
+            e.Data().RequestedOperation(DataPackageOperation::Move);
+
+            // The next thing that will happen:
+            //  * Another TerminalPage will get a TabStripDragOver, then get a
+            //    TabStripDrop
+            //    * This will be handled by the _other_ page asking the monarch
+            //      to ask us to send our content to them.
+            //  * We'll get a TabDroppedOutside to indicate that this tab was
+            //    dropped _not_ on a TabView.
+            //    * This we can't handle yet, and is the last point of TODO GH#5000
+        }
+    }
+
+    void TerminalPage::_onTabStripDragOver(winrt::Windows::Foundation::IInspectable /*sender*/,
+                                           winrt::Windows::UI::Xaml::DragEventArgs e)
+    {
+        // We must mark that we can accept the drag/drop. The system will never
+        // call TabStripDrop on us if we don't indicate that we're willing.
+        if (e.DataView().Properties().HasKey(L"windowId"))
+        {
+            e.AcceptedOperation(DataPackageOperation::Move);
+        }
+
+        // You may think to yourself, this is a great place to increase the
+        // width of the TabView artificially, to make room for the new tab item.
+        // However, we'll never get a message that the tab left the tab view
+        // (without being dropped). So there's no good way to resize back down.
+    }
+
+    // Method Description:
+    // - Called on the TARGET of a tab drag/drop. We'll unpack the DataPackage
+    //   to find who the tab came from. We'll then ask the Monarch to ask the
+    //   sender to move that tab to us.
+    winrt::fire_and_forget TerminalPage::_onTabStripDrop(winrt::Windows::Foundation::IInspectable /*sender*/,
+                                                         winrt::Windows::UI::Xaml::DragEventArgs e)
+    {
+        // Get the PID and make sure it is the same as ours.
+        if (const auto& pidObj{ e.DataView().Properties().TryLookup(L"pid") })
+        {
+            const auto pidStr{ winrt::unbox_value<winrt::hstring>(pidObj) };
+            uint32_t pidNum;
+            if (Utils::StringToUint(pidStr.c_str(), pidNum))
+            {
+                if (pidNum != GetCurrentProcessId())
+                {
+                    // The PID doesn't match ours. We can't handle this drop.
```

Comment:
Is it safe to check this on `DragOver` as well?

Reply:
I'd rather clue users in early

Reply:
perhaps with a message? Are we allowed to include arbitrary "YOU CANNOT DROP BECAUSE" text?

Reply:
> perhaps with a message? Are we allowed to include arbitrary "YOU CANNOT DROP BECAUSE" text

Perhaps. Our Drag drop UX is always gonna be a bit wack. I'll file a follow up, to investigate how that might be possible today.